packages/backend/src/graphql/resolvers/tenant.test.ts

```diff
@@ -0,0 +1,508 @@
+import { IocContract } from '@adonisjs/fold'
+import { AppServices } from '../../app'
+import { createTestApp, TestContainer } from '../../tests/app'
+import {
+  DeleteTenantMutationResponse,
+  Tenant,
+  TenantMutationResponse,
+  TenantsConnection,
+  WhoamiResponse
+} from '../generated/graphql'
+import { initIocContainer } from '../..'
+import { Config, IAppConfig } from '../../config/app'
+import { createTenant, generateTenantInput } from '../../tests/tenant'
+import { ApolloError, gql, NormalizedCacheObject } from '@apollo/client'
+import { getPageTests } from './page.test'
+import { truncateTables } from '../../tests/tableManager'
+import {
+  createHttpLink,
+  ApolloLink,
+  ApolloClient,
+  InMemoryCache
+} from '@apollo/client'
+import { setContext } from '@apollo/client/link/context'
+import { GraphQLErrorCode } from '../errors'
+import { Tenant as TenantModel } from '../../tenants/model'
+
+function createTenantedApolloClient(
+  appContainer: TestContainer,
+  tenantId: string
+): ApolloClient<NormalizedCacheObject> {
+  const httpLink = createHttpLink({
+    uri: `http://localhost:${appContainer.app.getAdminPort()}/graphql`,
+    fetch
+  })
+  const authLink = setContext((_, { headers }) => {
+    return {
+      headers: {
+        ...headers,
+        'tenant-id': tenantId
+      }
+    }
+  })
+
+  const link = ApolloLink.from([authLink, httpLink])
+
+  return new ApolloClient({
+    cache: new InMemoryCache({}),
+    link: link,
+    defaultOptions: {
+      query: {
+        fetchPolicy: 'no-cache'
+      },
+      mutate: {
+        fetchPolicy: 'no-cache'
+      },
+      watchQuery: {
+        fetchPolicy: 'no-cache'
+      }
+    }
+  })
+}
+
+describe('Tenant Resolvers', (): void => {
+  let deps: IocContract<AppServices>
+  let appContainer: TestContainer
+  let config: IAppConfig
+
+  beforeAll(async (): Promise<void> => {
+    deps = await initIocContainer({
+      ...Config,
+      databaseUrl: process.env.TENANT_TEST_DATABASE_URL as string
```

Comment:
Why do we need this separate db instead of just using `TEST_DATABASE_URL`?

Reply:
I added another test database to handle pagination tests correctly for tenants. Pagination tests need to truncate the tables between tests in order to work properly - if we do this on the same database as the other tests, that will break the ones that require the existence of an operator tenant to work, as it will have been deleted.

I realize that the pagination tests will be testing a scenario that shouldn't happen, but I feel like it makes sense to do it this way to make sure that pagination works correctly.

Reply:
I think you can get away with this by using e.g. `dbSchema` config instead. This is how I ended up solving this [(comment)](https://github.com/interledger/rafiki/pull/3206#discussion_r1911158951).